runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/state/WindmillStateInternals.java

```diff
@@ -82,7 +81,8 @@ public WindmillStateInternals(
     return key;
   }
 
-  private void persist(List<Future<WorkItemCommitRequest>> commitsToMerge, StateTable stateTable) {
+  private void persist(
+      List<Future<WorkItemCommitRequest>> commitsToMerge, CachingStateTable stateTable) {
     for (State location : stateTable.values()) {
       if (!(location instanceof WindmillState)) {
```

Comment:
since we have our own class now I think you could enforce this with typing instead of having to cast here